### PR TITLE
 railjson: add the new signaling model 

### LINKED ISSUE #2610: Integrate the new signaling model into RailJSON
## Description

Belongs to #1046

The new signaling simulator brings new requirements, and lifts others:
 - signals loose the `linked_detector` field, which used to be required for block delimiters
 - signals gain a `logical_signals` field. each logical signal is simulated independently, and has:
    - a `signaling_system` field
    - a `settings` field
    - a `next_signaling_systems` field

We decided on a two phase implementation:
 - introduce the new fields first
 - make the new fields mandatory and deprecate others second

see [the reference specification](https://dgexsolutions.github.io/osrd-website/fr/docs/reference/design-docs/signaling/)

## Acceptance criteria

- [x] The OpenAPI specification was updated with the optional added fields
- [x] The simulation backend can load the new fields
- [x] The python infrastructure generator library was updated with support for the new fields
- [x] The edition service can edit the new fields

### PR DESCRIPTION
**Pull request by Khoyo**
Related to #2610

### CODE REVIEW

File: editoast/src/schema/signal.rs [hunk 1]
```diff
@@ -28,6 +28,9 @@ pub struct Signal {
     #[derivative(Default(value = "400."))]
     pub sight_distance: f64,
     pub linked_detector: Option<String>,
+    pub signaling_system: String,
+    pub next_signaling_systems: Vec<String>,
+    pub settings: Vec<String>,
```
Comment (Khoyo) [inline editoast/src/schema/signal.rs]: This should be in a LogicalSignal struct 😅

File: api/osrd_infra/schemas/infra.py [hunk 2]
```diff
@@ -372,6 +372,14 @@ class TrackSection(BaseObjectTrait, GeometryLineTrait):
     )
 
 
+class LogicalSignal(BaseModel):
+    """A logical signal is what displays something, whereas a physical signal is a group of logical signals"""
+
+    signaling_system: str = Field(description="The signal's output signaling system")
+    next_signaling_systems: List[str] = Field(description="The list of allowed input signaling systems")
+    settings: List[str] = Field(description="A list of key value parameters, which are defined per signaling system")
```
Comment (flomonster) [inline api/osrd_infra/schemas/infra.py]:
```suggestion
    settings: Mapping[NonBlankStr, NonBlankStr] = Field(description="A list of key value parameters, which are defined per signaling system")
```

**Review by flomonster — CHANGES_REQUESTED** [review]
Few things:

- We should bump railjson version
- You shouldn't use raw `str` but `NonBlankStr`

**Review by multun — APPROVED** [review]

File: api/osrd_infra/schemas/infra.py (hunk 2 above)
Comment (Khoyo) [reply api/osrd_infra/schemas/infra.py]: Done

**Review by flomonster — APPROVED** [review]

**Pull request merged**